les/peer.go

```diff
@@ -945,6 +948,10 @@ func (p *clientPeer) freezeClient() {
 // Handshake executes the les protocol handshake, negotiating version number,
 // network IDs, difficulties, head and genesis blocks.
 func (p *clientPeer) Handshake(td *big.Int, head common.Hash, headNum uint64, genesis common.Hash, server *LesServer) error {
+	disableTxLookup := server.handler.blockchain.TxLookupLimit() != 0
```

Comment:
Can we put the "transaction available range" in the handshake? For example if the `server.handler.blockchain.TxLookupLimit()` is 10, then the transactions in `HEAD-10` to `HEAD` are still available.

I know with we still can't use this number meaningfully. However at least it's better then "disabled".

For example, if all the servers are unindexing the transactions in the future, we can still somehow send the requests to the server with maximum available range..

Reply:
The problem is that the client does not know the age of the tx, only the hash. So it cannot decide whether the server with the limited tx index is suitable for the particular request or not.

Reply:
Yes I understand. But if in the future, we enable the unindexing by default, at least we can try some servers.
So right now, we can always pick the les with unindexing disabled. But in the future, we can still try some servers. There is no downside.

Reply:
Ok, you're right. Now the handshake always includes an integer that is 0 if tx lookup is unlimited and N+1 if the limit is N blocks (a small safety margin is also subtracted).